github-actions/unified-status-check/src/main.ts

```diff
@@ -0,0 +1,202 @@
+import * as core from '@actions/core';
+import {context} from '@actions/github';
+import {types as graphqlTypes, params, optional, onUnion, query} from 'typed-graphqlify';
+import {Octokit} from '@octokit/rest';
+import {getAuthTokenFor, ANGULAR_ROBOT, revokeActiveInstallationToken} from '../../utils.js';
+import {CheckConclusionState, PullRequestState, StatusState} from '@octokit/graphql-schema';
+
+const unifiedStatusCheckName = 'Unified Status';
+
+async function main() {
+  /** A Github API instance. */
+  const github = new Octokit({auth: await getAuthTokenFor(ANGULAR_ROBOT)});
+
+  try {
+    /** Status matchers to ignore within the context of the action, always ignoring the actions own status. */
+    const ignored = [
+      unifiedStatusCheckName,
+      ...core.getMultilineInput('ignored', {trimWhitespace: true}),
+    ];
+    /** Status matchers which must match at least one of the current statuses . */
+    const required = core.getMultilineInput('required', {trimWhitespace: true});
+    /** The pull request triggering the event */
+    const pullRequest = (
+      await github.graphql<typeof PR_SCHEMA>(query(PR_SCHEMA).toString(), {
+        ...context.issue,
+      })
+    ).repository.pullRequest;
+    /** The status check rollup results. */
+    const statusChecks = pullRequest.commits.nodes[0].commit.statusCheckRollup;
+
+    const setStatus = async (
+      state: 'pending' | 'error' | 'failure' | 'success',
+      description?: string,
+    ) => {
+      await github.repos.createCommitStatus({
```

Comment:
One interesting thing I just realized. IIRC the Angular GitHub stored the status in Firebase right?

Problem is that calling `createCommitStatus` too often will quickly flood a commit with too many statuses, ultimately preventing more statuses to be created. I faced this in the presubmit status service when it was not debounced, so now a new status on the commit is only applied if the state changes.

Just calling this out- we might be able to hit this, if we update whenever statuses change. That can evolve pretty quickly.

Reply:
Added in a check for if we are just doing the same update on the status.